src/store.rs

```diff
@@ -438,6 +438,12 @@ impl<F: LurkField> IntoHashComponents<F> for ScalarPtr<F> {
 #[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
#293 contained a suggestion to interpret the first element of a `ScalarPtr` for the purpose of displaying which type of tag to expect. It would be great to edit the `impl<F> Display for ScalarPtr<F>` to implement it.

This would require interpreting the first field element of the `ScalarPtr`. Looking at `ExprTag`, which encodes some conventions used there, would be useful for this probing. 

Note that overall, not all field elements that can be interpreted as tags are in `ExprTag`. The comments on the `Expression` type in `store.rs`, the [spec](https://blog.lurk-lang.org/posts/circuit-spec/) are resources I've found useful to try to grasp this.

Ultimately, I'm just a newbie and @emmorais @porcuquine may help.